selfservice/hook/web_hook.go

```diff
@@ -307,9 +350,58 @@ func doHttpCall(method string, url string, as AuthStrategy, body io.Reader) erro
 
 	if err != nil {
 		return err
-	} else if resp.StatusCode >= 400 {
+	} else if resp.StatusCode >= http.StatusBadRequest {
```

Comment:
So we have two options here:

1. If the response is an error code we stop execution and try to understand the error, relaying it to the UI
2. If the response contains a special field (e.g. `abort_flow`) we interrupt the flow and display an error

Generally speaking I think calling the config `can_interrupt` is misleading, because an error will always interrupt the flow. What this feature is doing is to allow us to gracefully return validation errors to the user, as opposed to showing a 500 server error.

Reply:
I was thinking about it in a similar way and I've decided to go with HTTP status codes to be somewhat in sync with what the Kratos client will receive.

Option 1. Is a bit more logical from the hook perspective that 200s will mean success and will not interrupt the flow and any other status code will one way or the other cause errors.
Option 2. Is a bit cleaner on Kratos' side since it will explicitly indicate an event that is being triggered by the web hook reponse.

I can go with either one and do not have strong preferences. I guess it is your call then @aeneasr :-)